nautobot/extras/models/roles.py

```diff
@@ -0,0 +1,93 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from nautobot.core.fields import AutoSlugField
+from nautobot.core.models import BaseModel
+from nautobot.extras.fields import LimitedChoiceField
+from nautobot.extras.models import ChangeLoggedModel, CustomFieldModel, RelationshipModel
+from nautobot.extras.models.mixins import NotesMixin
+from nautobot.extras.querysets import ContentTypeRelatedQuerySet
+from nautobot.extras.utils import RoleModelsQuery, extras_features
+from nautobot.utilities.choices import ColorChoices
+from nautobot.utilities.fields import ColorField
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "webhooks",
+)
+class Role(BaseModel, ChangeLoggedModel, CustomFieldModel, RelationshipModel, NotesMixin):
+    # TODO(timizuo): Create abstract model for this because Role, Status and Tags
+    #  all contain similar fields
+
+    content_types = models.ManyToManyField(
+        to=ContentType,
+        related_name="statuses",
+        verbose_name="Content type(s)",
+        limit_choices_to=RoleModelsQuery(),
+        help_text="The content type(s) to which this status applies.",
+    )
+    name = models.CharField(max_length=50, unique=True)
+    color = ColorField(default=ColorChoices.COLOR_GREY)
+    slug = AutoSlugField(populate_from="name", max_length=50)
```

Comment:
IMO `http://localhost:8080/a-slug-used` is more url friendly than `http://localhost:8080/a%20slug%20used`

Reply:
I agree with both but I think we can leave this as-is and once we address #2834 (part of #1574) we can probably use the natural key in the URL instead of with spaces.